src/services/completions.ts

```diff
@@ -1031,7 +1031,7 @@ namespace ts.Completions {
             // Need to insert 'this.' before properties of `this` type, so only do that if `includeInsertTextCompletions`
             if (preferences.includeCompletionsWithInsertText && scopeNode.kind !== SyntaxKind.SourceFile) {
                 const thisType = typeChecker.tryGetThisTypeAt(scopeNode);
-                if (thisType) {
+                if (thisType && thisType.symbol.name !== "globalThis") {

```

Comment:
Is this reliable? Is there no way to name the symbol associated with a `thisType`?

Reply:
Um, not reliable, no. Reliable would be checking that `thisType.symbol !== typeChecker.getGlobalThisSymbol()` (and creating that function). Or making tryGetThisTypeAt not return globalThis sometimes &mdash; it's only used 3 places, and this one never wants `globalThis`.

I'm going to add the parameter.